tests/phpunit/tests/rest-api/rest-settings-controller.php

```diff
@@ -83,13 +83,60 @@ public function test_context_param() {
 	public function test_get_item_is_not_public_not_authenticated() {
 		$request  = new WP_REST_Request( 'GET', '/wp/v2/settings' );
 		$response = rest_get_server()->dispatch( $request );
-		$this->assertSame( 401, $response->get_status() );
+		$data     = $response->get_data();
+		$actual   = array_keys( $data );
+
+		$expected = array(
+			'description',
+			'language',
+			'page_for_posts',
+			'page_on_front',
+			'show_on_front',
+			'title',
+		);
+
+		sort( $expected );
+		sort( $actual );
+
+		$this->assertSame( 200, $response->get_status() );
+		$this->assertSame( $expected, $actual );
 	}
 
 	public function test_get_item_is_not_public_no_permission() {
```

Comment:
Function name is not longer accurate. Also, I think this test and the one above can be combined and a data provider can be used for the different user types.